diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SortRule.kt

```diff
@@ -0,0 +1,109 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.KtLint
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.CLASS_BODY
+import com.pinterest.ktlint.core.ast.ElementType.CONST_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
+import com.pinterest.ktlint.core.ast.ElementType.SEMICOLON
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.SORT_RULE
+import org.cqfn.diktat.ruleset.utils.*
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+
+class SortRule : Rule("sort-rule") {
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.isClassEnum() && node.hasChildOfType(CLASS_BODY))
+            sortEnum(node.findChildByType(CLASS_BODY)!!)
+        if (node.isCompanionObject() &&  node.hasChildOfType(CLASS_BODY))
+            sortProperty(node.findChildByType(CLASS_BODY)!!)
+    }
+
+    private fun sortProperty(node: ASTNode) {
+        val propertyList = node.getChildren(null)
+                .filter { it.elementType == PROPERTY }
+                .filter { it.hasChildOfType(MODIFIER_LIST) }
+                .filter { it.findChildByType(MODIFIER_LIST)!!.hasChildOfType(CONST_KEYWORD) }
+        if (propertyList.size <= 1)
+            return
+        val orderListOfList = createOrderListOfList(propertyList)
+        val sortedListOfList = mutableListOf<MutableList<ASTNode>>()
+        orderListOfList.forEach { nodesList ->
+            val sortList = nodesList.toMutableList()
+            sortList.sortBy { it.findChildByType(IDENTIFIER)!!.text }
+            sortedListOfList.add(sortList)
+        }
+        if (orderListOfList != sortedListOfList) {
+            SORT_RULE.warnAndFix(configRules, emitWarn, isFixMode,
```

Comment:
I think it's better to emit warning for every group of properties, this way it will be easier to navigate.